x-pack/legacy/plugins/ingest/server/libs/datasources.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { FrameworkUser } from './adapters/framework/adapter_types';
+import { Datasource } from './adapters/policy/adapter_types';
+
+export class DatasourcesLib {
+  private datasources: Datasource[] = [];
+
+  public async getByIDs(_user: FrameworkUser, ids: string[]): Promise<Datasource[]> {
+    return new Promise(resolve => {
```

Comment:
```suggestion
    return this.datasources.filter(ds => ids.includes(ds.id))
```

Reply:
Im happy to change that. This wa just a stub so I just threw it together, was not trying to be pandantic about quality here 😊